extensions/wikia/SpecialPromote/UploadVisualizationImageFromFile.class.php

```diff
@@ -57,7 +57,9 @@ public function checkWarnings(){
 	static public function UploadVerification($destName, $tempPath, &$error) {
 		$promoImage = PromoImage::fromPathname($destName);
 
-		if($promoImage->isValid() and ($promoImage->getCityId() == F::app()->wg->cityId)){
+		if ( ( $promoImage->isValid() && $promoImage->getCityId() == F::app()->wg->cityId ) ||
+			!PhalanxTitleBlock::checkTitle( Title::newFromText( $destName ), true )
+		) {
```

Comment:
You told me to move the check logic to this hook! 😭 
![Why?](https://ci.memecdn.com/289/2392289.jpg)